Lib/cu2qu/__init__.py

```diff
@@ -227,18 +226,22 @@ def curves_to_quadratic(curves, max_errors):
     """
 
     curves = [[complex(*p) for p in curve] for curve in curves]
-    num_curves = len(curves)
-    assert len(max_errors) == num_curves
+    assert len(max_errors) == len(curves)
 
-    splines = [None] * num_curves
     for n in range(1, MAX_N + 1):
-        splines = [cubic_approx_spline(c, n, e)
-                   for c, e in zip(curves, max_errors)]
-        if all(splines):
+        splines = []
+        for c, e in zip(curves, max_errors):
+            spline = cubic_approx_spline(c, n, e)
+            if spline is None:
+                break
+            splines.append(spline)
+        else:
+            # done. go home
             break
```

Comment:
So we never add `None` to `splines`, does that mean we never raise an `ApproxNotFoundError`? I suggest moving the return statement here and raising an error unconditionally at the end of the function, maybe with all of the input curves.